src/nft/components/bag/BagFooter.tsx

```diff
@@ -172,24 +190,55 @@ export const BagFooter = ({
   return (
     <FooterContainer>
       <Footer>
-        <FooterHeader gap="xs" warningText={!!warningText}>
-          <Row justify="space-between">
-            <div>
-              <ThemedText.HeadlineSmall>Total</ThemedText.HeadlineSmall>
-            </div>
-            <div>
-              <ThemedText.HeadlineSmall>
+        {shouldUsePayWithAnyToken && (
+          <CurrencyRow>
+            <Column gap="xs">
+              <ThemedText.SubHeaderSmall textAlign="start">
+                <Trans>Pay with</Trans>
+              </ThemedText.SubHeaderSmall>
+              <CurrencyInput>
+                <CurrencyLogo currency={inputCurrency} size="24px" />
+                <ThemedText.HeadlineSmall fontWeight={500} lineHeight="24px">
+                  {inputCurrency?.symbol}
+                </ThemedText.HeadlineSmall>
+                <ChevronDown size={20} color={theme.textSecondary} />
+              </CurrencyInput>
+            </Column>
+            <Column gap="xs">
+              <ThemedText.SubHeaderSmall marginBottom="4px" textAlign="end">
```

Comment:
if you move `textAlign="end"` up one level would all these components inherit it?

Reply:
valid